scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkBackend.scala

```diff
@@ -0,0 +1,226 @@
+        val evidence = SubTypes.fromSubType[A, A]
+        val rdd = rdd1
+      }
+
+    def toRDD[A1](r: R[A1]): RDD[A1] =
+      // this would be maybe slower, but safe:
+      r.rdd.map(r.evidence.toEv)
+    // or we can just cast
+    //r.rdd.asInstanceOf[RDD[A1]]
+
+    def empty(ctx: SparkContext): R[Nothing] =
+      R(ctx.emptyRDD[Nothing])
+
+    def fromIterable[A](ctx: SparkContext, iter: Iterable[A]): R[A] =
+      R(ctx.makeRDD(iter.toSeq, 1))
+
+    def merge[A1](left: R[A1], right: R[A1]): R[A1] =
+      R(toRDD(left) ++ toRDD(right))
+
+    implicit class PairR[K, V](val kv: R[(K, V)]) extends AnyVal {
+      def flatMapValues[U](fn: V => TraversableOnce[U]): R[(K, U)] =
+        R(toRDD(kv).flatMapValues(fn))
+      def mapValues[U](fn: V => U): R[(K, U)] =
+        R(toRDD(kv).mapValues(fn))
+    }
+  }
+
+  /**
+   * Convert a TypedPipe to an RDD
+   */
+  def plan(ctx: SparkContext, config: Config)(implicit ec: ExecutionContext): FunctionK[TypedPipe, R] =
+    Memoize.functionK(new Memoize.RecursiveK[TypedPipe, R] {
+      import TypedPipe._
+
+      def toFunction[A] = {
+        case (cp @ CounterPipe(_), rec) =>
+          // TODO: counters not yet supported
+          def go[A](p: CounterPipe[A]): R[A] = {
+            rec(p.pipe).map(_._1)
+          }
+          go(cp)
+        case (cp @ CrossPipe(_, _), rec) =>
+          def go[A, B](cp: CrossPipe[A, B]): R[(A, B)] =
+            rec(cp.viaHashJoin)
+          go(cp)
+        case (CrossValue(left, EmptyValue), rec) => rec(EmptyTypedPipe)
+        case (CrossValue(left, LiteralValue(v)), rec) =>
+          val op = rec(left) // linter:disable:UndesirableTypeInference
+          op.map((_, v))
+        case (CrossValue(left, ComputedValue(right)), rec) =>
+          rec(CrossPipe(left, right))
+        case (p: DebugPipe[a], rec) =>
+          // There is really little that can be done here but println
+          rec[a](p.input.map(DebugFn()))
+        case (EmptyTypedPipe, rec) =>
+          R.empty(ctx)
+        case (fk @ FilterKeys(_, _), rec) =>
+          def go[K, V](node: FilterKeys[K, V]): R[(K, V)] = {
+            val FilterKeys(pipe, fn) = node
+            rec(pipe).filter(FilterKeysToFilter(fn))
+          }
+          go(fk)
+        case (f @ Filter(_, _), rec) =>
+          def go[T](f: Filter[T]): R[T] = {
+            val Filter(p, fn) = f
+            rec[T](p).filter(fn)
+          }
+          go(f)
+        case (f @ FlatMapValues(_, _), rec) =>
+          def go[K, V, U](node: FlatMapValues[K, V, U]) = {
+            val fn = node.fn
+            rec(node.input).flatMapValues(fn)
+          }
+
+          go(f)
+        case (FlatMapped(prev, fn), rec) =>
+          val op = rec(prev) // linter:disable:UndesirableTypeInference
+          op.concatMap(fn)
+        case (ForceToDisk(pipe), rec) =>
+          rec(pipe).persist(StorageLevel.DISK_ONLY)
+        case (Fork(pipe), rec) =>
+          rec(pipe).persist(StorageLevel.MEMORY_ONLY)
```

Comment:
don't need to bother updating this here, but i'd leave these to DISK_ONLY until we can figure out how/when we can upgrade the state. (Spark also has the output of shuffles persisted, so we might be able to get the planner to realize when we should have shuffle data and not do the persisting then).